neo/Trie/MPT/MPTNode.cs

```diff
@@ -34,56 +41,36 @@ public MPTNode()
             Dirty = true;
         }
 
-        public byte[] Encode()
+        public virtual void Serialize(BinaryWriter writer)
         {
-            using (BinaryWriter writer = new BinaryWriter(new MemoryStream(), Encoding.UTF8, false))
-            {
-                writer.Write((byte)nType);
-                EncodeSpecific(writer);
-                writer.Flush();
-                return ((MemoryStream)writer.BaseStream).ToArray();
-            }
+            writer.Write((byte)Type);
         }
 
-        public abstract void EncodeSpecific(BinaryWriter writer);
+        public virtual void SerializeAsChild(BinaryWriter writer)
+        {
+            var hn = new HashNode(GetHash());
+            hn.SerializeAsChild(writer);
+        }
+
+        public abstract void Deserialize(BinaryReader reader);
 
-        public static MPTNode Decode(byte[] data)
+        public static unsafe MPTNode DeserializeFromByteArray(byte[] data)
         {
-            if (data is null || data.Length == 0)
-                return null;
+            if (data is null || data.Length < 1) return null;
 
-            MPTNode node;
-            using (BinaryReader reader = new BinaryReader(new MemoryStream(data, false), Encoding.UTF8))
+            fixed (byte* pointer = data)
```

Comment:
Why use unsafe? we can use a `MemoryStream`

Reply:
done